ci/code_checks.sh

```diff
@@ -15,6 +15,7 @@ RET=$(($RET + $?)) ; echo $MSG "DONE"
 
 MSG='Checking mypy... ' ; echo $MSG
 mypy dask_ml/metrics
+mypy --no-strict-optional dask_ml/preprocessing
```

Comment:
Ideally we do these all at once. I think you can pass multiple paths.

Why do you need no-strict-optional? mypy recommends against disabling strict-optional: https://mypy.readthedocs.io/en/stable/kinds_of_types.html#disabling-strict-optional-checking

Reply:
I get these errors:
```
 mypy dask_ml/preprocessing/
dask_ml/preprocessing/label.py:151: error: Item "None" of "Optional[Any]" has no attribute "categories"
dask_ml/preprocessing/label.py:167: error: Item "None" of "Optional[Any]" has no attribute "categories"
dask_ml/preprocessing/label.py:169: error: Item "None" of "Optional[Any]" has no attribute "ordered"
dask_ml/preprocessing/label.py:186: error: Item "None" of "Optional[Any]" has no attribute "categories"
dask_ml/preprocessing/label.py:187: error: Item "None" of "Optional[Any]" has no attribute "ordered"
dask_ml/preprocessing/_encoders.py:267: error: Argument 2 to "zip" has incompatible type "Optional[Any]"; expected "Iterable[Any]"
```

`self.dtype_: Optional[pd.CategoricalDtype] = None`  makes mypy infer `dtype_` as `Any

Reply:
OK, thanks. It seems like we'll need a check that `isinstance(self._dtype, CategoricalDtype)`  under that branch. There's an implicit assumption that type passed to `transform` matches the type passed to `fit`.

Reply:
Adding `if self.dtype_ is not None:` solves this issue.